packages/shared/src/shorthand-properties.ts

```diff
@@ -0,0 +1,136 @@
+  ],
+  gridArea: ['gridRowStart', 'gridColumnStart', 'gridRowEnd', 'gridColumnEnd'],
+  gridColumn: ['gridColumnStart', 'gridColumnEnd'],
+  gridGap: ['gridColumnGap', 'gridRowGap'],
+  gridRow: ['gridRowStart', 'gridRowEnd'],
+  gridTemplate: ['gridTemplateColumns', 'gridTemplateRows', 'gridTemplateAreas'],
+  inset: ['top', 'right', 'bottom', 'left'],
+  listStyle: ['listStyleType', 'listStylePosition', 'listStyleImage'],
+  margin: ['marginTop', 'marginRight', 'marginBottom', 'marginLeft'],
+  mask: ['maskImage', 'maskMode', 'maskRepeat', 'maskPosition', 'maskClip', 'maskOrigin', 'maskSize', 'maskComposite'],
+  maskBorder: [
+    'maskBorderSource',
+    'maskBorderMode',
+    'maskBorderSlice',
+    'maskBorderWidth',
+    'maskBorderOutset',
+    'maskBorderRepeat',
+  ],
+  offset: ['offsetPosition', 'offsetPath', 'offsetDistance', 'offsetRotate', 'offsetAnchor'],
+  outline: ['outlineWidth', 'outlineStyle', 'outlineColor'],
+  overflow: ['overflowX', 'overflowY'],
+  padding: ['paddingTop', 'paddingRight', 'paddingBottom', 'paddingLeft'],
+  pause: ['pauseBefore', 'pauseAfter'],
+  placeContent: ['alignContent', 'justifyContent'],
+  placeItems: ['alignItems', 'justifyItems'],
+  placeSelf: ['alignSelf', 'justifySelf'],
+  rest: ['restBefore', 'restAfter'],
+  scrollMargin: ['scrollMarginTop', 'scrollMarginRight', 'scrollMarginBottom', 'scrollMarginLeft'],
+  scrollPadding: ['scrollPaddingTop', 'scrollPaddingRight', 'scrollPaddingBottom', 'scrollPaddingLeft'],
+  scrollPaddingBlock: ['scrollPaddingBlockStart', 'scrollPaddingBlockEnd'],
+  scrollPaddingInline: ['scrollPaddingInlineStart', 'scrollPaddingInlineEnd'],
+  scrollSnapMargin: ['scrollSnapMarginTop', 'scrollSnapMarginRight', 'scrollSnapMarginBottom', 'scrollSnapMarginLeft'],
+  scrollSnapMarginBlock: ['scrollSnapMarginBlockStart', 'scrollSnapMarginBlockEnd'],
+  scrollSnapMarginInline: ['scrollSnapMarginInlineStart', 'scrollSnapMarginInlineEnd'],
+  scrollTimeline: ['scrollTimelineSource', 'scrollTimelineOrientation'],
+  textDecoration: ['textDecorationLine', 'textDecorationStyle', 'textDecorationColor'],
+  textEmphasis: ['textEmphasisStyle', 'textEmphasisColor'],
+  transition: ['transitionProperty', 'transitionDuration', 'transitionTimingFunction', 'transitionDelay'],
+}
+
+const longhands = Object.values(shorthandProperties).reduce((a, b) => [...a, ...b], [])
+
+/**
+ * Get the property priority: the higher the priority, the higher the resulting
+ * specificity of the atom. For example, if we had:
+ *
+ * ```tsx
+ * import { css } from '@pandacss/dev';
+ *
+ * css({
+ *   backgroundColor: 'blue',
+ *   background: 'red',
+ * })
+ *
+ * ```
+ *
+ * and so the more specific selector would win
+ *
+ * Taken from
+ * @see https://github.com/callstack/linaria/blob/049a4ccb77e29f3628353352db21bd446fa04a2e/packages/atomic/src/processors/helpers/propertyPriority.ts
+ */
+export function getPropertyPriority(property: string) {
+  return longhands.includes(property) ? 2 : 1
```

Comment:
Is this sufficient complexity, or should there be more priority levels? I notice that in StyleX they define four different layers of priority for normal properties ([code](https://github.com/facebook/stylex/blob/c97421056c819f042271d717eee8ce5c1a4c1669/packages/shared/src/utils/property-priorities.js#L755-L766)), from highest to lowest priority:
1. Longhand physical properties (e.g. `border-bottom-color`)
2. Longhand logical properties (e.g. `border-block-end-color`)
3. Shorthand versions of longhand-only properties (e.g. `border-bottom`)
4. Shorthand versions of shorthand properties (e.g. `border`)

Reply:
can you give me an example with `css({ ... })` where the current implementation would be problematic ?